app/src/main/java/de/stephanlindauer/criticalmaps/handler/ChooseGpxFileHandler.java

```diff
@@ -0,0 +1,22 @@
+package de.stephanlindauer.criticalmaps.handler;
+
+import android.content.Intent;
+
+import de.stephanlindauer.criticalmaps.R;
+import de.stephanlindauer.criticalmaps.fragments.SettingsFragment;
+import de.stephanlindauer.criticalmaps.vo.RequestCodes;
+
+public class ChooseGpxFileHandler {
+    private final SettingsFragment activity;
+
+    public ChooseGpxFileHandler(SettingsFragment activity) {
+        this.activity = activity;
+    }
+
+    public void openChooser() {
+        Intent chooseTrack = new Intent(Intent.ACTION_OPEN_DOCUMENT);
+        chooseTrack.setType("*/*");
```

Comment:
I tested the persistence across reboots and it works. But should it?
According to the docs only grants to URIs requested with `Intent.FLAG_GRANT_PERSISTABLE_URI_PERMISSION` should be able to be persisted...
Am I missing something here?

Reply:
see GpxUtils line 9, called in SettingsFragment line 119. I don't know, if that's the best or correct way to persist permissions, i found it in the docs.

Reply:
Yes, I think the part to persist the permission in GpxUtils is fine (even though I think we do not need the `FLAG_GRANT_WRITE_URI_PERMISSION` since we do not modify the file).
What I was talking about is the part where you create the chooser intent. According to the docs you should only be allowed to later on persist the grant, if you got the URI via an intent that has the `Intent.FLAG_GRANT_PERSISTABLE_URI_PERMISSION` set via `Intent.setFlags()`:

> Only URI permissions granted with Intent#FLAG_GRANT_PERSISTABLE_URI_PERMISSION can be persisted.

On the other hand the example code in the docs - which I think you followed - does not mention this.
Since after testing everything works as intended, I would just leave it like that.